ArduCopter/mode_acro_heli.cpp

```diff
@@ -29,24 +29,36 @@ void Copter::ModeAcro_Heli::run()
     float target_roll, target_pitch, target_yaw;
     float pilot_throttle_scaled;
 
-    // Tradheli should not reset roll, pitch, yaw targets when motors are not runup, because
-    // we may be in autorotation flight.  These should be reset only when transitioning from disarmed
-    // to armed, because the pilot will have placed the helicopter down on the landing pad.  This is so
-    // that the servos move in a realistic fashion while disarmed for operational checks.
-    // Also, unlike multicopters we do not set throttle (i.e. collective pitch) to zero so the swash servos move
-    
-    if (motors->init_targets_on_arming()) {
-        attitude_control->reset_rate_controller_I_terms();
-        attitude_control->set_attitude_target_to_current_attitude();
+    // Tradheli should not reset roll, pitch, yaw targets when motors are not runup while flying, because
+    // we may be in autorotation flight.  This is so that the servos move in a realistic fashion while disarmed
+    // for operational checks. Also, unlike multicopters we do not set throttle (i.e. collective pitch) to zero
+    // so the swash servos move.
+
+    if (!motors->armed()) {
+        // Motors should be Stopped
+        motors->set_desired_spool_state(AP_Motors::DESIRED_SHUT_DOWN);
+    } else {
+        // heli will not let the spool state progress to THROTTLE_UNLIMITED until motor interlock is enabled
+        motors->set_desired_spool_state(AP_Motors::DESIRED_THROTTLE_UNLIMITED);
     }
 
-    // clear landing flag above zero throttle
-    if (motors->armed() && motors->get_interlock() && motors->rotor_runup_complete() && !ap.throttle_zero) {
-        set_land_complete(false);
+    if (motors->get_spool_mode() == AP_Motors::SHUT_DOWN) {
```

Comment:
can't this be a switch case ?

Reply:
in fact the whole block can be factorize as use multiple time accross modes

Reply:
possibly, basically it's the same pattern as drfit...